examples/chip-tool/commands/pairing/PairingCommand.cpp

```diff
@@ -161,9 +161,8 @@ CHIP_ERROR PairingCommand::PairWithMdns(NodeId remoteId)
 
 CHIP_ERROR PairingCommand::Unpair(NodeId remoteId)
 {
-    CHIP_ERROR err = CurrentCommissioner().UnpairDevice(remoteId);
-    SetCommandExitStatus(err);
-    return err;
+    mCurrentFabricRemover = Platform::MakeUnique<Controller::CurrentFabricRemover>(&CurrentCommissioner());
+    return mCurrentFabricRemover->CurrentFabricRemove(remoteId, &mCurrentFabricRemoveCallback);
```

Comment:
The right way to use a CurrentFabricRemover is to deallocate it if `CurrentFabricRemove` returns failure, else deallocate it in the callback, no?

Reply:
Yes, you are right.
However, if it allocate using MakeUnique (std:unique_ptr), the automatic allocation is released at the end of use. (OpenCommissioningWindowCommand is also used similarly.)